### ISSUE TITLE
Add Pinpoint costing info to notifications table

### ISSUE THREAD

**Issue by sastels** [cds-snc/notification-planning-core#438]
# Description

As a member of the Notify team, I need access to SMS cost data so that I can ensure that we are spending our money wisely.

## WHY are we building?

SNS gave us a nightly dump of SMS usage and costs that we imported into QuickSight to analyze. Unfortunately Pinpoint does not have this functionality.

## WHAT are we building?

Pinpoint includes the costing info in the delivery receipts, so we can pull it from there and add it to the notifications. While we're at it we should add a few additional useful fields. We get a json blob similar to:
```
{
    "eventType": "TEXT_DELIVERED",
    "eventVersion": "1.0",
    "eventTimestamp": 1729013399221,
    "isFinal": true,
    "originationPhoneNumber": “+*******”,
    "destinationPhoneNumber": “+*******”,
    "isoCountryCode": "CA",
    "mcc": "302",
    "mnc": "510",
    "carrierName": "Videotron Ltd.",
    "messageId": "c3819844-1f2d-4ff0-a521-7c0f42392215",
    "messageRequestTimestamp": 1729013362726,
    "messageEncoding": "GSM",
    "messageType": "TRANSACTIONAL",
    "messageStatus": "DELIVERED",
    "messageStatusDescription": "Message has been accepted by phone",
    "totalMessageParts": 1,
    "totalMessagePrice": 0.00581,
    "totalCarrierFee": 0.00767
}
```

## VALUE created by our solution

We can once again keep track of where we're spending money on SMS.

# Acceptance Criteria

- [x] Performance test in staging shows little to no impact on database performance
- [x] Determine if any of these fields have/require an index (**update**: no)
- [x] Fields added to the notifications and notification_history table:
  - totalMessagePrice
  - totalCarrierFee
  - isoCountryCode
  - carrierName
  - messageEncoding
  - originationPhoneNumber (note: this is the Notify number)

# Out of Scope
The modifications to the quicksight queries are not in scope for this task.

# QA Steps

- [x] Send yourself an sms
- [x] Verify the values for the new fields for your notification.

**Comment by sastels** [comment]
PR https://github.com/cds-snc/notification-api/pull/2326

**Comment by ben851** [comment]
Perf testing required in staging before merging

**Comment by sastels** [comment]
Baselining today

**Comment by sastels** [comment]
Waiting on baselining until after the k8s upgrade to 3.11

**Comment by sastels** [comment]
Will merge and test in staging today!

**Comment by sastels** [comment]
merged into staging.
Did some small tests to see new fields populated in notifications table
load test started at 11:45

**Comment by sastels** [comment]
- Send rate similar to before merging PR (about 5600 emails / min sustained)
- can't load test SMS because the dryrun causes throttling errors :/
- send SMS to myself as well as a 20-sized bulk job (added more long codes to staging to support) and all worked, new fields populated in database

**Comment by jimleroyer** [comment]
We merged into staging last week. Checks in the database were done. The database looks fine and a rollercoaster test was performed up to expectations. SMS works fine as before.

**Comment by jimleroyer** [comment]
@jimleroyer to qa!

**Comment by jimleroyer** [comment]
QA: new fields are present and looks good. Good job! 👏
<img width="1534" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN0NHQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--621a3597e09ed657d1c2e54e4b1b965f3a760ae8/image.png" alt="image.png" />
